playground/backend/internal/cloud_bucket/precompiled_objects.go

```diff
@@ -0,0 +1,283 @@
+			CloudPath:   object.CloudPath,
+			Name:        object.Name,
+			Description: object.Description,
+			Type:        object.Type,
+		})
+	}
+	sdkCategories.Categories = append(sdkCategories.Categories, &category)
+}
+
+// getPrecompiledObjectsDirs finds "directories" with precompiled objects
+func (cd *CloudStorage) getPrecompiledObjectsDirs(ctx context.Context, targetSdk pb.Sdk, bucket *storage.BucketHandle) (map[string]bool, error) {
+	prefix := targetSdk.String()
+	if targetSdk == pb.Sdk_SDK_UNSPECIFIED {
+		prefix = ""
+	}
+	it := bucket.Objects(ctx, &storage.Query{
+		Prefix: prefix,
+	})
+	objectDirs := make(map[string]bool, 0)
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Bucket(%q).Objects: %v", BucketName, err)
+		}
+		path := attrs.Name
+		if isPathToPrecompiledObject(path) {
+			precompiledObjectDir := filepath.Dir(path)
+			_, ok := objectDirs[precompiledObjectDir]
+			if !ok {
+				objectDirs[precompiledObjectDir] = true
+			}
+		}
+	}
+	return objectDirs, nil
+}
+
+// appendPrecompiledObject add precompiled object to the common structure of precompiled objects
+func appendPrecompiledObject(objectInfo ObjectInfo, sdkToCategories *SdkToCategories, pathToObject string, categoryName string) {
+	sdkName := getSdkName(pathToObject)
+	categoryToPrecompiledObjects, ok := (*sdkToCategories)[sdkName]
+	if !ok {
+		(*sdkToCategories)[sdkName] = make(CategoryToPrecompiledObjects, 0)
+		categoryToPrecompiledObjects = (*sdkToCategories)[sdkName]
+	}
+	objects, ok := categoryToPrecompiledObjects[categoryName]
+	if !ok {
+		categoryToPrecompiledObjects[categoryName] = make(PrecompiledObjects, 0)
+		objects = categoryToPrecompiledObjects[categoryName]
+	}
+	objectInfo.CloudPath = pathToObject
+	objectInfo.Name = filepath.Base(pathToObject)
+	categoryToPrecompiledObjects[categoryName] = append(objects, objectInfo)
+}
+
+// getFileFromBucket receives the file from the bucket by its name
+func (cd *CloudStorage) getFileFromBucket(ctx context.Context, pathToObject string, extension string) ([]byte, error) {
+	client, err := storage.NewClient(ctx, option.WithoutAuthentication())
+	if err != nil {
+		return nil, fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*Timeout)
+	defer cancel()
+
+	bucket := client.Bucket(BucketName)
+
+	filePath := getFullFilePath(pathToObject, extension)
+	rc, err := bucket.Object(filePath).NewReader(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Object(%q).NewReader: %v", filePath, err)
+	}
+	defer rc.Close()
+
+	data, err := ioutil.ReadAll(rc)
+	if err != nil {
+		return nil, fmt.Errorf("ioutil.ReadAll: %v", err)
+	}
+	return data, nil
+}
+
+// getFileExtensionByFileSdk get extension of the file with code by the sdk name
+func getFileExtensionByFileSdk(precompiledObjectPath string) string {
+	sdk := strings.Split(precompiledObjectPath, "/")[0]
```

Comment:
Is there a predefined constant for "/"? If no, I think we should put it into our constants